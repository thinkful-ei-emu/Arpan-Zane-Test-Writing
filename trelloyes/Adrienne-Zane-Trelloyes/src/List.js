import React from 'react';
import Card from './Card.js';
import STORE from './store.js';

export default function List(props) {
   const individualCards = props.cards.map((card) => (
    <Card key={card.id} title={card.title} content={card.content} />
  ));
  console.log(individualCards);
  console.log(props.cards);
  return (
  <section className="List">
    <header className="List-header">{props.header}
    </header>
    <div className="List-cards">
      {individualCards}
    </div>
  </section>
  );
}