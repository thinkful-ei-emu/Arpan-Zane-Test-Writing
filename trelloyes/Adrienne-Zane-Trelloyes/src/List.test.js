import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';
import STORE from './store';

const list ={
    id: '1',
    header: 'First list',
    cardIds: [ 'a', 'b', 'e', 'f', 'g', 'j', 'l', 'm' ],
  };

describe('<List/>',()=>{
    it('renders without crashing',()=>{
        const div= document.createElement('div');

        ReactDOM.render((<List key= {list.id} title={list.header} cards={list.cardIds.map(id => STORE.allCards[id])} />), div);

        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders this UI as Expected',()=>{
        const tree = renderer.create((<List key= {list.id} title={list.header} cards={list.cardIds.map(id => STORE.allCards[id])} />)).toJSON();
        expect(tree).toMatchSnapshot();
    });
});