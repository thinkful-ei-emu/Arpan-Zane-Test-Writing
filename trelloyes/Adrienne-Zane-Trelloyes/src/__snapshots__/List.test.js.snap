// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List/> renders this UI as Expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  />
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        Fifth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        Sixth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        Seventh card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        Tenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        Twelfth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        Thirteenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
